modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/object/entries/frontend/data/set/view/table/ObjectEntriesTableFDSView.java

```diff
@@ -256,19 +273,55 @@ else if (Objects.equals(
 
 			fdsTableSchemaField = stringFDSTableSchemaField;
 		}
-		else if (Objects.equals(dbType, ObjectFieldConstants.DB_TYPE_DATE)) {
+		else if (Objects.equals(
+					businessType, ObjectFieldConstants.BUSINESS_TYPE_DATE)) {
+
 			DateFDSTableSchemaField dateFDSTableSchemaField =
 				new DateFDSTableSchemaField();
 
 			dateFDSTableSchemaField.setFieldName(fieldName);
-			dateFDSTableSchemaField.setFormat("short");
+			dateFDSTableSchemaField.setFormat(
+				_getFormatJSONObject(businessType, null));
 			dateFDSTableSchemaField.setLabel(label);
 			dateFDSTableSchemaField.setLocalizeLabel(localizeLabel);
 
 			fdsTableSchemaBuilder.add(dateFDSTableSchemaField);
 
 			fdsTableSchemaField = dateFDSTableSchemaField;
 		}
+		else if (Objects.equals(
+					businessType,
+					ObjectFieldConstants.BUSINESS_TYPE_DATE_TIME)) {
+
+			DateTimeFDSTableSchemaField dateTimeFDSTableSchemaField =
+				new DateTimeFDSTableSchemaField();
+
+			dateTimeFDSTableSchemaField.setFieldName(fieldName);
+
+			User user = null;
+
+			try {
+				user = _userLocalService.getUser(
+					PrincipalThreadLocal.getUserId());
+			}
+			catch (PortalException portalException) {
+				if (_log.isWarnEnabled()) {
+					_log.warn(portalException);
+				}
+			}
+
+			dateTimeFDSTableSchemaField.setFormat(
+				_getFormatJSONObject(
+					businessType,
+					ObjectFieldSettingUtil.getTimeZoneId(
+						objectFieldSettings, user)));
```

Comment:
@murilostodolni please add a empty line after `dateTimeFDSTableSchemaField.setFormat(`, as this code depends on the `user` variable initialization..